app/src/main/java/com/feedhenry/securenativeandroidtemplate/mvp/views/BaseAppView.java

```diff
@@ -0,0 +1,48 @@
+package com.feedhenry.securenativeandroidtemplate.mvp.views;
+
+import android.app.Fragment;
+import android.content.Context;
+import android.widget.Toast;
+
+import com.feedhenry.securenativeandroidtemplate.mvp.components.ProgressDialogHelper;
+
+/**
+ * A base implementation for the AppView interface.
+ */
+
+public abstract class BaseAppView implements AppView {
+
+    private Fragment fragment;
+    private ProgressDialogHelper progressDialogHelper;
+
+    public BaseAppView(Fragment fragment) {
+        this.fragment = fragment;
+        this.progressDialogHelper = new ProgressDialogHelper();
+
+    }
+
+    @Override
+    public void showLoading() {
+        Context ctx = getContext();
+        if (ctx != null) {
+            this.progressDialogHelper.showProgress(ctx);
+        }
+    }
+
+    @Override
+    public void hideLoading() {
+        this.progressDialogHelper.hideProgress();
+    }
+
+    @Override
+    public void showError(String errorMessage) {
+        Context ctx = getContext();
+        if (ctx != null) {
+            Toast.makeText(ctx, errorMessage, Toast.LENGTH_LONG).show();
```

Comment:
Maybe we could look at using [Snackbars](https://developer.android.com/training/snackbar/action.html) to display errors. They allow setting an action button along with the message so it might be useful for doing retries or any other actions if something fails.

Reply:
yeah, I will try that